src/main/java/org/sagebionetworks/web/server/servlet/filter/HtmlInjectionFilter.java

```diff
@@ -0,0 +1,373 @@
+
+              if (includeBotHtml) {
+                dataModel.put(
+                  BOT_BODY_HTML_KEY,
+                  crawlFilter.getThreadHtml(thread, discussionThreadContent)
+                );
+              }
+            } else {
+              // index information about the synapse entity
+              EntityBundleRequest bundleRequest = new EntityBundleRequest();
+              bundleRequest.setIncludeEntity(true);
+              bundleRequest.setIncludeAnnotations(true);
+
+              EntityBundle bundle = synapseClient.getEntityBundleV2(
+                entityId,
+                bundleRequest
+              );
+              Entity entity = bundle.getEntity();
+              String description = "";
+              try {
+                WikiPageKey key = new WikiPageKey();
+                key.setOwnerObjectId(entityId);
+                key.setOwnerObjectType(ObjectType.ENTITY);
+                key.setWikiPageId(null);
+                WikiPage rootPage = synapseClient.getWikiPage(key);
+                description = CrawlFilter.getPlainTextWiki(entityId, rootPage);
+              } catch (Exception e) {}
+
+              dataModel.put(PAGE_TITLE_KEY, entity.getName());
+              dataModel.put(PAGE_DESCRIPTION_KEY, description);
+
+              if (includeBotHtml) {
+                BotHtml botHtml = crawlFilter.getEntityHtml(bundle);
+                dataModel.put(BOT_HEAD_HTML_KEY, botHtml.getHead());
+                dataModel.put(BOT_BODY_HTML_KEY, botHtml.getBody());
+              }
+            }
+          } else if (path.startsWith("/Search")) {
+            // index all projects
+            String searchQueryRawValue = uri.substring(uri.indexOf(":") + 1);
+            SearchQuery query = SearchQueryUtils.getDefaultSearchQuery();
+            try {
+              query =
+                EntityFactory.createEntityFromJSONString(
+                  URLDecoder.decode(searchQueryRawValue, "UTF-8"),
+                  SearchQuery.class
+                );
+            } catch (Exception e) {
+              query.setQueryTerm(
+                Collections.singletonList(searchQueryRawValue)
+              );
+            }
+            String queryTerm = query.getQueryTerm().get(0);
+            if (queryTerm != null && queryTerm.trim().length() > 0) {
+              dataModel.put(
+                PAGE_TITLE_KEY,
+                "Searching for: " + query.getQueryTerm().get(0)
+              );
+            }
+
+            if (includeBotHtml) {
+              dataModel.put(
+                BOT_BODY_HTML_KEY,
+                crawlFilter.getAllProjectsHtml(query)
+              );
+            }
+          } else if (path.startsWith("/TeamSearch")) {
+            TeamSearch place = new TeamSearch(placeToken);
+            dataModel.put(
+              PAGE_TITLE_KEY,
+              SEARCHING_FOR_TEAM + place.getSearchTerm()
+            );
```

Comment:
Will there always be team search term or should this also omit "Searching for Team: " if there is no search term?

Reply:
will fix